db/db_utils.go

```diff
@@ -25,7 +26,11 @@ var (
 )
 
 type SessionFactory interface {
+	// GetSession uses the global environment's context to get a session and database.
 	GetSession() (db.Session, db.Database, error)
+	// GetContextSession uses the provided context to get a session and database.
+	// This is needed for operations that need control over the passed in context.
+	GetContextSession(ctx context.Context) (db.Session, db.Database, error)
```

Comment:
I didn't include this in the comment but we could add a TODO comment pointing to [DEVPROD-11824](https://jira.mongodb.org/browse/DEVPROD-11824) and make a comment there about this function (I'm planning on picking it up and I know about it, the TODO/comment is just if we want to provide support if someone else picks it up).

The reason we need it is because in go driver v2, setting the timeout is now handled by cancelling a context (context.withtimeout) rather than setting an option- so our db queries need a context. So technically dead code right now so I could remove it if the reviewer wants!

Reply:
I think it's worth still adding a TODO since the ticket isn't assigned to you, but if you want to assign it that works too!

also nit: "passed-in context"

Reply:
Also will we still need GetSession, or will we remove this as part of DEVPROD-11824 as well?

Reply:
I think we might be able to remove it. I'll add a TODO comment!